src/v/cluster_link/task.cc

```diff
@@ -51,6 +51,12 @@ class task::runner {
     void set_task_interval(ss::lowres_clock::duration interval) {
         vlog(
           _task->logger().trace, "set_task_interval called with {}", interval);
+        if (!_timer.armed()) {
+            // can only happen when a task is currently running;
+            // its continuation will rearm the timer
+            return;
+        }
+
```

Comment:
this looks like it makes the interval update susceptible to being ignored in a race because _timer.armed isn't strictly tied to the life cycle of the runner.

for example, if set_task_interval is called while run_task() is executing then the timer won't be armed, even though the task hasn't necessarily been stopped.

Reply:
> if set_task_interval is called while run_task() is executing then the timer won't be armed

It will. Each call to `run_task()` is followed by a `_timer.arm()` call.
To avoid confusion I probably should remove the `_timer.cancel();` line below.

> this looks like it makes the interval update susceptible to being ignored

The interval for the next timer arm is stored in `task`, not in `task::runner`, and is updated by `task::set_run_interval`. This update affects `_task->_run_interval`, which will be used next time `task::runner` arms the timer.

Reply:
> It will. Each call to run_task() is followed by a _timer.arm() call.

isn't the key point here _followed by_ the rearming of the timer? while `run_task` is executing (which may block, and thus be interleaved with other calls like `set_task_interval`) the timer is unarmed. further, the `finally` continuation that rearms the timer shouldn't be considered to run atomically with `run_task`.

Reply:
i see, thanks. i must have been confused by the naming. it's not clear why task::set_task_interval doesn't seem to set the task interval, but rather that is the run_interval set by something else. anyway, lgtm.